src/components/AddBook.jsx

```diff
@@ -1,66 +1,111 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import { useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
 import { v4 as uuidv4 } from 'uuid';
+import { ProgressSpinner } from 'primereact/progressspinner';
 import { addBook } from '../redux/books/bookSlice';
 
-const AddBook = () => {
+const AddBook = ({ entity, toast }) => {
   const [title, setTitle] = useState('');
   const [author, setAuthor] = useState('');
+  const [category, setCategory] = useState('Fiction');
+  const [status, setStatus] = useState(true);
   const dispatch = useDispatch();
 
-  const scrollToTop = () => {
-    window.scrollTo(0, 0);
+  const selectCategory = (e) => {
+    setCategory(e.target.value);
   };
 
   const addABook = () => {
+    if (title === '' || author === '') {
+      toast.fire({
+        icon: 'error',
+        title: 'All fields are required to add a book',
+      });
+    }
+
     if (title !== '' && author !== '') {
+      setStatus(false);
       const book = {
         item_id: uuidv4(),
         title,
         author,
-        category: 'Sci-fi',
+        category,
       };
       dispatch(addBook(book));
-      scrollToTop();
       setTitle('');
       setAuthor('');
     }
   };
 
+  useEffect(() => {
+    if (entity.msg) {
+      setStatus(true);
+    }
+  }, [entity.msg]);
+
   return (
-    <section id="form-section" className="mt-5">
-      <h3>Add New Book</h3>
-      <form className="flex flex-col md:flex-row gap-5">
-        <div className="form-control flex gap-5">
+    <section id="form-section" className="mt-10 border-t-2 pt-5">
+      <h3 className="text-2xl font-bold">Add New Book</h3>
+      <form className="">
+        <div className="form-control grid md:grid-cols-3 lg:grid-cols-4 gap-x-5 gap-y-5">
           <input
             type="text"
-            className="p-2 border focus:outline-none w-[12.5rem] md:w-[27rem] lg:w-[35rem]"
+            className="p-3 md:p-2 border focus:outline-none col-span-1"
             placeholder="Book title"
             value={title}
             onChange={(e) => setTitle(e.target.value)}
             required
           />
           <input
             type="text"
-            className="p-2 border focus:outline-none w-[12.5rem] md:w-[27rem] lg:w-[35rem]"
+            className="p-3 md:p-2 border focus:outline-none"
             placeholder="Author"
             value={author}
             onChange={(e) => setAuthor(e.target.value)}
             required
           />
```

Comment:
- [ ] [REQUIRED] For this field where a book gets added, you need to follow the exact styles as used in the zeplin design. Kindly change the font, color, font weight, and every other property the (Add new book, book title, category, add book button) has.
### Your Design
<img width="1440" alt="Screenshot 2022-11-08 at 18 45 56" src="https://user-images.githubusercontent.com/58692590/200640636-6314b9b9-795f-4937-9eaf-42c485cb80ca.png">

### Zeplin Design
<img width="1019" alt="Screenshot 2022-11-08 at 18 59 48" src="https://user-images.githubusercontent.com/58692590/200640972-5c7b4984-7150-4a64-9476-4498b9c98b63.png">

If you click on the `ADD NEW BOOK` for example, you will get a property of it on the right side of your screen. Kindly apply these properties for every content of your design.
### For example
<img width="1440" alt="Screenshot 2022-11-08 at 19 01 45" src="https://user-images.githubusercontent.com/58692590/200641588-cceb9307-1b25-463d-93ef-6c70e1803f1f.png">